flake.nix

```diff
@@ -7,6 +7,10 @@
       path = builtins.path { path = ./example; filter = path: _: baseNameOf path == "flake.nix"; };
     };
 
+    # NOTE: this is a lib factory, as it takes: { pkgs, lib }
+    # so before using we need to pass it to pkgs.callPackage
+    lib = import ./nix/lib.nix;
```

Comment:
not sure if it's common for the `lib` output to be a function but `nix flake check` didn't complain

for this to be a function we'd have add nixpkgs as flake input

Reply:
We could use `inputs.nixpkgs.lib` but since the submodule below passes `pkgs` as well, as specialArgs, we necessarily need some perSystem context. @roberth is there an idiomatic way here?

Reply:
I wonder if it would be better to expose this via an overlay:

```nix
final: _: {
  services-flake = {
    lib = import ./nix/lib.nix { pkgs = final; };
  };
}
```

or if you want to give it a bit more thought, we could not expose it at all and I'd import the file directly via:

```nix
import "${services-flake}/nix/lib.nix" { inherit pkgs; }
```

Reply:
Cool, I’ll update the PR accordingly

Reply:
Thanks! I will merge it once you make the change.